samza-core/src/main/java/org/apache/samza/coordinator/leaderelection/LeaderElector.java

```diff
@@ -28,7 +30,7 @@
    *
    * @return True, if caller is chosen as a leader through the leader election process. False, otherwise.
    */
-  boolean tryBecomeLeader();
+  boolean tryBecomeLeader(LeaderElectorListener leaderElectorListener);
```

Comment:
Will add documentation.
I think we discussed at some point that we want all the api be Async. 
We don't need the boolean anymore, I will remove it.